core/TypePtr.h

```diff
@@ -73,30 +73,27 @@ class TypePtr final {
     };
     tagged_storage store;
 
-    // Top bit indicates if value is inlined into pointer.
-    // We use a 0 in top bit to indicate not inlined so that nullptr (which has counter value 0) is naturally viewed as
+    // We use a 0 to indicate not inlined so that nullptr (which has counter value 0) is naturally viewed as
     // 'inlined'.
-    static constexpr tagged_storage NOT_INLINED_MASK = 0x8000000000000000;
-    static constexpr tagged_storage TAG_MASK = 0x7FFF000000000007;
+    static constexpr tagged_storage NOT_INLINED_MASK = 0x100;
+    // We use a 8-bit tag mask so that tags can be efficiently extracted via
+    // movzx on x86.
+    static constexpr tagged_storage TAG_MASK = 0xFF;
```

Comment:
How do you feel about giving these leading `0`s, so that they look like 16-bit constants?
```suggestion
    static constexpr tagged_storage NOT_INLINED_MASK = 0x0100;
    // We use a 8-bit tag mask so that tags can be efficiently extracted via
    // movzx on x86.
    static constexpr tagged_storage TAG_MASK = 0x00FF;
```